### DISCUSSION TITLE
transition-group not work

### DISCUSSION THREAD

**Discussion by ibrahim-developper** [quasarframework/quasar#11327] [General - Components / Directives / etc] [answered]
i want to use transition group between two component

`

      <transition-group
        appear
        enter-active-class="animated fadeIn"
        leave-active-class="animated fadeOut"
      >

        <register-with-email v-if="account_type == 'email'" key="rgemail" />

        <register-with-phone-number v-else key="rgphone" />

      </transition-group>

`

and i'm add the plugin fadeIn and fadeOut inside quasar.conf.js

`
    animations: [
      'fadeIn',
      'fadeOut'
    ]
`
but it's not make any animation

`

  "dependencies": {
    "@quasar/cli": "^1.2.2",
    "@quasar/extras": "^1.11.4",
    "core-js": "^3.6.5",
    "firebase": "^9.2.0",
    "quasar": "^2.2.2",
    "vue-i18n": "^9.0.0",
    "vuex": "^4.0.1"
  }

`
**Thank you.**

**Comment by pdanpdan** [comment]
You don't have a group, you have a simple transition.

**Reply by ibrahim-developper** [reply]
i'm try simple transition but the same
`

      <transition
        appear
        enter-active-class="animated fadeIn"
        leave-active-class="animated fadeOut"
      >

        <register-with-email v-if="account_type == 'email'"  />

        <register-with-phone-number v-else  />

      </transition>

`

**Reply by ibrahim-developper** [reply]
https://user-images.githubusercontent.com/52491316/141653016-836db8d1-74da-46ab-9020-dd6ca15e7fc6.mov

this the transition

**Comment by hawkeye64** [comment]
Shouldn't this be brought up in the [vue-next](https://github.com/vuejs/vue-next) issues (as they don't have discussions turned on)?

**Reply by ibrahim-developper** [reply]
no

**Comment by rstoenescu** [comment, marked as answer]
The transitions work just fine.

It's just that you expect the transition to take place simultaneously, while the two states to occupy the same real-estate on screen, which is not what those transitions do. (Same thing can be seen [on Vue docs](https://v3.vuejs.org/guide/transitions-enterleave.html#transition-modes). You'd need additional work done by yourself on the CSS due to how Vue `<transition>` works. Like absolute positioning those elements that you transition OR using the `mode="out-in"` prop+value.

**Reply by rstoenescu** [reply]
Alternatively, you could use QSlideTransition instead. https://quasar.dev/vue-components/slide-transition

**Reply by ibrahim-developper** [reply]
Thank you alot ♥️🙏
